lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -676,22 +689,42 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
         if not self.plugin_manager.check_plugin_manager_ready():
             return
 
+        if self.upgrade_in_progress:
```

Comment:
Moved this upwards as, from here down to the end of `_config_changed`, changes should not be accepted during upgrade.